website/content/posts/a-taste-of-policies.md

```diff
@@ -0,0 +1,232 @@
+      code: github_certificate.gh_workflow_ref
+    msg: 'Ref "refs/heads/master" not in allowed list: ["refs/heads/spam"]'
+  - metadata:
+      code: github_certificate.gh_workflow_repository
+    msg: 'Repository "lcarva/festoji" not in allowed list: ["spam/spam"]'
+  - metadata:
+      code: github_certificate.gh_workflow_trigger
+    msg: 'Trigger "push" not in allowed list: ["spam"]'
+{{</code>}}
+
+As you can see, the Enterprise Contract provides a comprehensive list of all the identified
+violations.
+
+## Bring Your Own Policy Rules
+
+In addition to the policy rules provide by the Enterprise Contract, it is also possible to use your
+own policy rules. The [festoji-policies](https://github.com/lcarva/festoji-policies) git repository
+illustrates this. It contains a very simple layout:
+
+{{<code bash>}}
+$ tree
+.
+├── LICENSE
+└── policies
+    └── github_slsa_provenance.rego
+
+2 directories, 2 files
+{{</code>}}
+
+The important bits are in the rego file:
+
+{{<code rego>}}
+# METADATA
+# title: GitHub SLSA Provenance
+# description: >-
+#   Verify SLSA Provenance created in GitHub meets requirements.
+package festoji.policies.github_slsa_provenance
+
+import future.keywords.contains
+import future.keywords.if
+import future.keywords.in
+
+# METADATA
+# title: Materials
+# description: Verify SLSA Provenance materials are correct.
+# custom:
+#   short_name:  materials
+#   failure_msg: Unexpected materials
+deny contains result if {
+	some att in input.attestations
+	match := [material |
+		some material in att.statement.predicate.materials
+		material.uri == "git+https://github.com/lcarva/festoji@refs/heads/master"
+	]
+	count(match) == 0
+	result := "Unexpected materials"
+}
+{{</code>}}
+
+This defines a single policy rule that verifies the materials section of the SLSA Provenance contain
+the expected git repository. The Enterprise Contract relies on rego annotations to provide
+additional information about each policy rules. See the
+[docs](https://enterprisecontract.dev/docs/ec-policies/authoring.html) for more information.
+
+Let's add this rule to our previous policy:
+
+{{<code yaml>}}
+---
+identity:
+  subjectRegExp: >-
+    https:\/\/github\.com\/(slsa-framework\/slsa-github-generator|lcarva\/festoji)\/
+  issuer: https://token.actions.githubusercontent.com
+
+sources:
+  - policy:
+      - github.com/enterprise-contract/ec-policies//policy/lib
+      - github.com/enterprise-contract/ec-policies//policy/release
+    ruleData:
+      allowed_gh_workflow_repos:
+        - lcarva/festoji
+      allowed_gh_workflow_refs:
+        - refs/heads/master
+      allowed_gh_workflow_names:
+        - Package
+      allowed_gh_workflow_triggers:
+        - push
+  # A new policy source group uses the custom policy rules.
+  - policy:
+    - github.com/lcarva/festoji-policies//policies
+
+configuration:
+  include:
+    - github_certificate
+    - github_slsa_provenance  # Also specify which custom policy rule to include.
```

Comment:
Not related to this PR, but this does highlight the janky-ness of the global config.

Reply:
Yeah, that's how I came across it :sweat_smile: